### PR TITLE
Move object change monitoring to the trigger-handling code

### PR DESCRIPTION
**Pull request by RobertSzefler**
Moves object change monitoring from the babysitter code to the more general event processing place.

### CODE REVIEW

File: src/robusta/integrations/kubernetes/base_triggers.py [hunk 1]
```diff
@@ -152,3 +169,30 @@ def build_execution_event(
             obj=obj,
             old_obj=old_obj,
         )
+
+    def check_change_filters(self, execution_event: ExecutionBaseEvent):
```
Comment (arikalon1) [inline src/robusta/integrations/kubernetes/base_triggers.py:173]: I think this function should only be called with `KubernetesAnyChangeEvent`

We can change the event type to `KubernetesAnyChangeEvent`, or we can add an `if isinstance(...)` at the beginning, and return `true` without doing anything on the wrong type

**Review by arikalon1 — APPROVED** [review]
very nice work!
left some minor comments

File: src/robusta/integrations/kubernetes/base_triggers.py (hunk 1 above)
Comment (RobertSzefler) [reply src/robusta/integrations/kubernetes/base_triggers.py:173]: Moved to `K8sBaseTrigger`

File: docs/tutorials/playbook-track-changes.rst [hunk 2]
```diff
@@ -33,10 +33,11 @@ Add the following YAML to the ``customPlaybooks`` Helm value:
     customPlaybooks:
     - triggers:
         - on_deployment_update: {}
+            change_filters:
+              ignore: [] # You can specify any fields here that you don't want to monitor
```
[the remark this answers is not in the document]
Comment (RobertSzefler) [reply docs/tutorials/playbook-track-changes.rst:37]: Added

File: playbooks/robusta_playbooks/babysitter.py [hunk 3]
```diff
@@ -18,30 +18,17 @@
...
-        "spec",
-    ]
-    omitted_fields: List[str] = [
-        "status",
-        "metadata.generation",
-        "metadata.resourceVersion",
-        "metadata.managedFields",
-        "spec.replicas",
-    ]
+    # TODO should this be moved to trigger code in PlaybooksEventHandlerImpl.handle_trigger as well?
```
[the remark this answers is not in the document]
Comment (RobertSzefler) [reply playbooks/robusta_playbooks/babysitter.py:31]: Removed

File: src/robusta/core/playbooks/playbooks_event_handler_impl.py [hunk 4]
```diff
@@ -52,6 +52,11 @@ def handle_trigger(self, trigger_event: TriggerEvent) -> Optional[Dict[str, Any]
                 execution_event = None
                 try:
                     execution_event = fired_trigger.build_execution_event(trigger_event, sink_findings, build_context)
+                    if hasattr(fired_trigger, "check_change_filters"):
```
[the remark this answers is not in the document]
Comment (RobertSzefler) [reply src/robusta/core/playbooks/playbooks_event_handler_impl.py:55]: Changed

File: src/robusta/integrations/kubernetes/base_triggers.py [hunk 5]
```diff
@@ -152,3 +169,30 @@ def build_execution_event(
...
+        """Check if the trigger should run considering the configuration of change_filters.
+
+        Sets appropriate fields related to filtered object data on the execution_event in
+        order for them to be accessible downstream in playbooks."""
+
+        if not self.change_filters:
+            execution_event.obj_filtered = execution_event.obj
+            execution_event.old_obj_filtered = execution_event.old_obj_filtered
+            execution_event.filtered_diffs = execution_event.obj.diff(execution_event.old_obj_filtered)
```
[the remark this answers is not in the document]
Comment (RobertSzefler) [reply src/robusta/integrations/kubernetes/base_triggers.py:182]: It's only executed if the user defines `change_filters` (see the `if` in line 179)

[the remark this answers is not in the document]
Comment (RobertSzefler) [reply src/robusta/integrations/kubernetes/base_triggers.py:182]: Changed to `[]`

File: src/robusta/core/playbooks/base_trigger.py [hunk 6]
```diff
@@ -19,7 +22,29 @@ def get_event_description(self) -> str:
...
+]
+
+
 class BaseTrigger(DocumentedModel):
+    change_filters: Dict[str, List[str]] = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.change_filters is None:
```
[the remark this answers is not in the document]
Comment (RobertSzefler) [reply src/robusta/core/playbooks/base_trigger.py]: Moved

File: src/robusta/integrations/kubernetes/base_triggers.py (hunk 1 above)
Comment (RobertSzefler) [reply src/robusta/integrations/kubernetes/base_triggers.py:173]: Added an `isinstance` check

**Pull request merged**